### PR TITLE
Unify struct definitions

### PR DESCRIPTION
**Pull request by akallabeth**
* Change enum/struct definitions so that compilers (clang, gcc) do not emit warnings (reserved identifier, e.g. name starts with `_`)
* Fix some cyclic include madness
* Fix problems with server builds (some functions were called with wrong arguments)

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: server/Windows/CMakeLists.txt [hunk 1]
```diff
@@ -89,7 +89,7 @@ endif()
 
 
 if(NOT CMAKE_WINDOWS_VERSION STREQUAL "WINXP")
-	set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} d3d11 dxgi dxguid)
+	set(${MODULE_PREFIX}_LIBS ${${MODULE_PREFIX}_LIBS} d3d11 dxgi)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply server/Windows/CMakeLists.txt:92]: @mfleisz builds without quite fine (e.g. no linking errors)

**Review by mfleisz — APPROVED** [review]

**Pull request merged**